### PR TITLE
LPD-5260 Add Backend Test for IndexerReindexer

### PR DESCRIPTION
**Pull request by joshuacords**
https://liferay.atlassian.net/browse/LPD-5260
Subtask: https://liferay.atlassian.net/browse/LPD-25881

It was difficult to what needed to be tested exactly. From what I could tell https://liferay.atlassian.net/browse/LPS-183661 made logical changes to ReindexIndexReindexerBackgroundTaskExecutor and those are what need to be tested. I don't think the background tasks were part of this so I don't see the need to add testing there.

@gustavolimav, let me know if I'm missing something.

**Author:** @joshuacords

### CODE REVIEW

**Comment by joshuacords** [comment]
ci:test:sf

**Comment by joshuacords** [comment]
ci:test:search

File: modules/apps/portal-search/portal-search-test/src/test/java/com/liferay/portal/search/internal/background/task/ReindexIndexReindexerBackgroundTaskExecutorTest.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
```
Comment (gustavolimav) [inline modules/apps/portal-search/portal-search-test/src/test/java/com/liferay/portal/search/internal/background/task/ReindexIndexReindexerBackgroundTaskExecutorTest.java:2]: 2024?

**Review by gustavolimav — CHANGES_REQUESTED** [review]
In general, the tests look great. Excellent work, Joshua! Just simple changes

Regarding the point about background tasks, I agree with your observation. Since the background tasks weren't originally part of the scope, there's no need to include additional testing for them in this PR. Good catch!

File: modules/apps/portal-search/portal-search-test/src/test/java/com/liferay/portal/search/internal/background/task/ReindexIndexReindexerBackgroundTaskExecutorTest.java (hunk 1 above)
Comment (joshuacords) [reply modules/apps/portal-search/portal-search-test/src/test/java/com/liferay/portal/search/internal/background/task/ReindexIndexReindexerBackgroundTaskExecutorTest.java:2]: Nice catch!

**Comment by joshuacords** [comment]
ci:test:sf

**Comment by joshuacords** [comment]
ci:test:search

**Comment by joshuacords** [comment]
Ready for your review @gustavolimav

**Review by gustavolimav — APPROVED** [review]
LGTM! 🚀

**Comment by joshuacords** [comment]
ci:forward

**Comment by gustavolimav** [comment]
ci:forward

**Pull request closed without merging**